### PR TITLE
[471716] Make the operators left-associative

### PR DESCRIPTION
**Pull request by jponge**
Operators are more naturally left-associative than right-associative.
The previous implementation was right-associative by mistake.

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=471716

File: src/main/java/fr/insalyon/citi/golo/compiler/ParseTreeToGoloIrVisitor.java [hunk 1]
```diff
@@ -1005,16 +1005,19 @@ public Object visit(ASTAdditiveExpression node, Object data) {
     Context context = (Context) data;
     node.childrenAccept(this, context);
     BinaryOperation current = null;
-    ExpressionStatement left;
-    ExpressionStatement right = null;
     List<String> symbols = node.getOperators();
-    Collections.reverse(symbols);
+    LinkedList<ExpressionStatement> statements = new LinkedList<>();
```
Comment (yloiseau) [inline src/main/java/fr/insalyon/citi/golo/compiler/ParseTreeToGoloIrVisitor.java]: Code duplication with multiplicative expression :)

File: src/main/java/fr/insalyon/citi/golo/compiler/ParseTreeToGoloIrVisitor.java [hunk 2]
```diff
@@ -1069,14 +1076,18 @@ public Object visit(ASTOrExpression node, Object data) {
     Context context = (Context) data;
     node.childrenAccept(this, context);
     BinaryOperation current = null;
-    ExpressionStatement left;
-    ExpressionStatement right = null;
-    for (int i = 0; i < node.count(); i++) {
-      if (right == null) {
-        right = (ExpressionStatement) context.objectStack.pop();
+    LinkedList<ExpressionStatement> statements = new LinkedList<>();
```
Comment (yloiseau) [inline src/main/java/fr/insalyon/citi/golo/compiler/ParseTreeToGoloIrVisitor.java]: code duplication with and expression :)

File: src/main/java/fr/insalyon/citi/golo/compiler/ParseTreeToGoloIrVisitor.java [hunk 3]
```diff
@@ -1088,14 +1099,18 @@ public Object visit(ASTOrIfNullExpression node, Object data) {
...
```
Comment (yloiseau) [inline src/main/java/fr/insalyon/citi/golo/compiler/ParseTreeToGoloIrVisitor.java]: idem...

File: src/main/java/fr/insalyon/citi/golo/compiler/ParseTreeToGoloIrVisitor.java (hunk 1 above)
Comment (jponge) [inline src/main/java/fr/insalyon/citi/golo/compiler/ParseTreeToGoloIrVisitor.java]: Yup :smile:

**Comment by jponge** [comment]
@yloiseau Here are some refactorings. Not perfect due to missing polymorphism in the AST classes, but avoids the gross of duplication.

**Comment by yloiseau** [comment]
:+1:

**Pull request merged**